Ibmmqconnector

hasithajayasundara opened the pull request:
**ibm-mq-connector for wso2 esb**

hasithajayasundara committed:
ibmmqconnector

hasithajayasundara committed:
Updated readme.md

hasithajayasundara committed:
Added topic based queueing

hasithajayasundara committed:
Seperated the topic and queue publish

hasithajayasundara committed:
log.error added

hasithajayasundara committed:
name change

hasithajayasundara committed:
Rename

hasithajayasundara committed:
Added customized connection pooling in WebSphere MQ

hasithajayasundara committed:
removed .idea

hasithajayasundara committed:
removed unwanted files

hasithajayasundara committed:
cleaned project

hasithajayasundara committed:
Changed the sample proxy configuration

nadeeshaan commented on src/test/resources/testng.xml:
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<!--
+ ~  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Please correct the license header to 2017 license header

hasithajayasundara replied:
Done.!

nadeeshaan commented on src/main/assembly/assemble-connector.xml:
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Please correct the license header to 2017 license header

hasithajayasundara replied:
Done.!

nadeeshaan commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,122 @@
+import java.util.Collection;
+import java.util.Vector;
+
+/**
```
Remove the current Java Doc and add a descriptive one

hasithajayasundara replied:
Done.!

nadeeshaan commented on src/main/java/org/wso2/carbon/esb/connector/MQConstants.java:
```diff
@@ -0,0 +1,32 @@
+package org.wso2.carbon.esb.connector;
+
```
Remove the current Java Doc and add a descriptive one

hasithajayasundara replied:
Done.!

nadeeshaan commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java:
```diff
@@ -0,0 +1,72 @@
+import org.wso2.carbon.connector.core.ConnectException;
+
+/**
+ * Created by hasitha on 6/30/17.
```
Remove the current Java Doc and add a descriptive one

hasithajayasundara committed:
Added 2017 licence headers and javadoc comments

hasithajayasundara committed:
Changed readme.md

hasithajayasundara committed:
Removed unwanted files

hasithajayasundara committed:
Added jars folder

hasithajayasundara committed:
Added client-keystore.jks

hasithajayasundara committed:
Updated readme.md

hmrajas commented on src/main/java/org/wso2/carbon/esb/connector/MQConfiguration.java:
```diff
@@ -0,0 +1,239 @@
+package org.wso2.carbon.esb.connector;
+
```
Please add wso2 licence header to each file

hasithajayasundara replied:
Done.!

hasithajayasundara committed:
Added licence headers,Combined queue and topic publish,Updated readme.md

hasithajayasundara committed:
Variable changes

hasithajayasundara committed:
Updated readme.md

hasithajayasundara committed:
Updated readme.md

hasithajayasundara committed:
Licence header changes

hasithajayasundara committed:
Added group id property and changed readme.md

hasithajayasundara committed:
Added message type parameter

hasithajayasundara committed:
Added method level comments

hasithajayasundara committed:
Added report message and datagram message

hasithajayasundara committed:
Added priority parameter

hasithajayasundara committed:
reademe changes

hasithajayasundara committed:
reademe changes

hasithajayasundara committed:
Added endpoint attribute

hasithajayasundara committed:
Removed unnecessary parameters

hasithajayasundara committed:
Removed request message

hasithajayasundara committed:
Added report to soap

hasithajayasundara committed:
Refactor the code for fatagra,request and report

hasithajayasundara committed:
Switch change

hasithajayasundara committed:
Readme updated

hasithajayasundara committed:
updated without ssl

hasithajayasundara committed:
Seperated topic and queue

hasithajayasundara committed:
updated readme.md

hasithajayasundara committed:
Message listening removed

hasithajayasundara committed:
Listener changed

hasithajayasundara committed:
datagram method changed

hasithajayasundara committed:
Changed listener

hasithajayasundara committed:
Added access mode property

hasithajayasundara committed:
Access mode changes

hasithajayasundara committed:
Readme updated

hasithajayasundara committed:
Readme updated

hasithajayasundara committed:
Message ID retrieve added

hasithajayasundara committed:
Added reconnecting options

hasithajayasundara committed:
Resolved listener conflicts

hasithajayasundara committed:
Readme updated

hasithajayasundara committed:
Readme updated

hasithajayasundara committed:
Readme updated

hasithajayasundara committed:
Minor change for loop

hasithajayasundara committed:
Minor change for loop

hasithajayasundara committed:
loop change

hasithajayasundara committed:
SSL ciphersuit adjusted

hasithajayasundara committed:
Reademe updated

hasithajayasundara committed:
Merge pull request #1 from hasithajayasundara/ibmmqconnector

Ibmmqconnector

hasithajayasundara committed:
override change

hasithajayasundara committed:
Parameter changes

djkevincr commented on org.wso2.carbon.esb.connector.ibmmqconnector.iml:
```diff
@@ -0,0 +1,624 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
We need remove this intellij idea generated files.

hasithajayasundara replied:
Done

djkevincr commented on pom.xml:
```diff
@@ -0,0 +1,376 @@
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+
```
Please remove this unnecessary new lines.

hasithajayasundara replied:
done

djkevincr commented on pom.xml:
```diff
@@ -0,0 +1,376 @@
+                    </instructions>
+                </configuration>
+            </plugin>
+
```
Please remove this unnecessary new lines.

hasithajayasundara replied:
done

djkevincr commented on src/main/assembly/filter.properties:
```diff
@@ -0,0 +1,3 @@
+product.name=WSO2 ESB
+product.version=4.9.0
+
```
Please remove this unnecessary new lines.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConfiguration.java:
```diff
@@ -0,0 +1,398 @@
+import java.util.Arrays;
+import java.util.List;
+
+import static com.ibm.mq.constants.CMQC.*;
```
Please remove this wild card imports, do this imports separately.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConfiguration.java:
```diff
@@ -0,0 +1,398 @@
+        }
+    }
+
+    public int getmaxConnections() {
```
Please use Javadoc proper standards to document all the public methods.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConfiguration.java:
```diff
@@ -0,0 +1,398 @@
+    private String qManger;
+    private String topicName;
+    private String topicString;
+    private String queue;
```
Please document all the parameters.

hasithajayasundara committed:
Delete .iml

hasithajayasundara committed:
Deleted .iml

hasithajayasundara committed:
Merge pull request #2 from hasithajayasundara/ibmmqconnector

Ibmmqconnector for loop

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+import java.io.FileInputStream;
+import java.security.KeyStore;
+import java.util.*;
+import java.util.concurrent.*;
```
Please remove these wild card imports.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+public class MQConnectionBuilder {
+
+    private static final Log logger = LogFactory.getLog(MQConnectionBuilder.class);
+
```
Please remove these unnecessary new lines.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+    private static final Log logger = LogFactory.getLog(MQConnectionBuilder.class);
+
+    private MQQueueManager queueManager;
+
```
Please remove these unnecessary new lines.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+    private MQConfiguration config;
+
+    MQConnectionBuilder(MessageContext msg) {
+
```
Please remove these unnecessary new lines.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+
+    private MQConfiguration config;
+
+    MQConnectionBuilder(MessageContext msg) {
```
Shall we change the method signature to MQConnectionBuilder(MQConfiguration config); and extract MQConfiguration from message context from earlier step.

hasithajayasundara committed:
Remove unnecessary changes

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+        channelList.add(0, config.getChannel());
+
+        if (queueManager == null || !queueManager.isConnected()) {
+            queueManager = getQueueManager(config.getHost(), config.getChannel(), config.getPort());
```
Please improve logging for,
[1] retry with retrying interval.
[2] Successful queue manager return.
[3] Unsuccessful queue manager return.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+
+    private static final Log logger = LogFactory.getLog(MQConnectionBuilder.class);
+
+    private MQQueueManager queueManager;
```
Builder classes usually not maintain state inside class. Shall we remove this class variable and use utility methods inside the this builder class.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+            initializeQmanager.get(2, TimeUnit.SECONDS);
+            logger.info("Queue manager connection established for " + host + " " + port + " " + channel);
+        } catch (InterruptedException e) {
+            return null;
```
Please use proper logging for these cases, if logging is not needed you may add comment saying this is expected scenario and we will return 'null', 'null' will be properly handled at higher layer.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+/**
+ * Start connection with IBM MQ queue manager
+ */
+public class MQConnectionBuilder {
```
Shall we rename this to MQConnectionUtils and use all public static methods?

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+     * Terminate connection with queue manager
+     */
+    public void closeConnection() {
+        try {
```
closeQueueManager(MQQueueManager queueManager);

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+    /**
+     * Terminate connection with queue manager
+     */
+    public void closeConnection() {
```
closeQueueManager(MQQueueManager queueManager);

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+    /**
+     * Setup customized connection pool for caching
+     */
+    MQSimpleConnectionManager customizedPool(long timeout, int maxConnections, int maxunusedConnections) {
```
Please add this as a private method.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+        return customizedPool;
+    }
+
+    public MQConfiguration getConfig() {
```
getConfig(MessageContext context)

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java:
```diff
@@ -0,0 +1,218 @@
+    /**
+     * Creating SSLContext for ssl connection
+     */
+    public SSLContext createSSLContext() {
```
Please add this as static method.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+*/
+package org.wso2.carbon.esb.connector;
+
+import com.ibm.mq.*;
```
Please remove all the wild card imports and do imports separately.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.ibm.mq.constants.CMQC.*;
```
Please remove all the wild card imports and do imports separately.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+public class MQPublishQueue extends AbstractConnector {
+
+    MQConnectionBuilder connectionBuilder;
+    MQConfiguration config;
```
Having MQConnectionBuilder, MQConfiguration at class level will not work for concurrent cases. Concurrently invoking will ended up using wrong configuration and connections if parameters are different.  We need to have some sort pooling mechanism to reuse IBM-MQ connections. Either we need to provide pooling mechanism from our side to handle this or reuse connection pooling mechanism from IBM-MQ client library. It s better if we could use IBM-MQ client library, if such available from client library.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+            }
+
+            //access queue manager
+            if(connectionBuilder==null || connectionBuilder.getManager()==null) {
```
if( - This is not formatted properly.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    public void connect(MessageContext messageContext) throws ConnectException {
+        try {
+
+            log.info("Message received at ibm-mq connector");
```
May be we should use debug here, otherwise we ll be flooded with these info logs.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+
+            //access queue manager
+            if(connectionBuilder==null || connectionBuilder.getManager()==null) {
+                connectionBuilder = new MQConnectionBuilder(messageContext);
```
First create MQConfiguration from messageContext properties and then pass that to utility method to MQConnectionUtils.getQueueManaget(MQConfiguration config) to retrieve queue managed from the pool.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+            }
+
+            if (queue == null || mqMessage == null) {
+                log.error("Error in publishing message");
```
Please make this log more informative, what is the exact reason failed publishing the message. Noticed you have returned null for queue, mqMessage for exceptional cases, at this point you have no idea behind the reason for this. This is not a good practice. Also you have already logged the exception at lower layer.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+            if (queue == null || mqMessage == null) {
+                log.error("Error in publishing message");
+            } else {
+                log.info("queue initialized");
```
This should be a debug log.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    /**
+     * Initialize queue
+     */
+    MQQueue setQueue(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
```
This should be private static method. You should pass the connManager returned from the pool.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+            queue = manager.accessQueue(config.getQueue(), CMQC.MQOO_OUTPUT);
+        } catch (MQException e) {
+            log.error("Error creating queue " + e);
+            return null;
```
Can we continue the flow getting the exception from here? Is this null case handled properly at higher layer. Or do we need to throw/propagate this to higher layer?

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    /**
+     * Create datagram or request MQMessage
+     */
+    MQMessage buildDatagramOrReplyMessage(MQConfiguration config, String queueMessage) {
```
These methods should be private.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
```
This should be error, log.error("Error creating Datagram/Reply mq message", e);

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    /**
+     * Create request MQMesaage
+     */
+    MQMessage buildRequestMessageandGetResponse(MQConfiguration config, String queueMessage) {
```
This method should be private.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
```
log.error("Error creating Request/Responce mq message", e);

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
+        }
+        return mqMessage;
```
I think for all these message types, it s better to throw the error to higher layer and from there it should be wrapped as ConnectionException. Returning message as empty and handle that as a generic one, one thing we loose information from lower layer why it is failed and this is not a best practice.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    /**
+     * Get reply message
+     */
+    void getReplyMessge(final MQQueue replyQueue, MQConfiguration config) {
```
I think can get rid of this method by pointing, IBM-MQ inbound to reply queue. Handling Listening part a connector in not good IMO.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    /**
+     * Create report MQMesaage
+     */
+    MQMessage buildReportMessage(MQConfiguration config, String queueMessage) {
```
This should be private static method.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    /**
+     * Get report message
+     */
+    void getReportMessage(final MQQueue replyQueue, MQConfiguration config) {
```
I think we can get rid of this method by pointing, ibm-mq inbound to reply queue. Handling Listening part inside a connector in not good IMO.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    /**
+     * generate report details
+     */
+    String reportStr(int code) {
```
This should be private static.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+    /**
+     * Initialize reply queue
+     */
+    MQQueue setReplyQueue(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
```
This should be private static.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java:
```diff
@@ -0,0 +1,386 @@
+            queue = manager.accessQueue(config.getreplyQueue(), CMQC.MQRC_READ_AHEAD_MSGS);
+        } catch (MQException e) {
+            log.error("Error creating reply queue" + e);
+            return null;
```
Returning null and handle error as generic, we will be loosing information why this got failed when you handle the null at higher layer. It s better if we could throw this exception to higher layer and handle over there.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java:
```diff
@@ -0,0 +1,134 @@
+ */
+public class MQPublishTopic extends AbstractConnector {
+
+    MQConnectionBuilder connectionBuilder;
```
This will not work for the concurrent cases. Please see related comments for queue, same goes here as well.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java:
```diff
@@ -0,0 +1,134 @@
+                    CMQC.MQTOPIC_OPEN_AS_PUBLICATION, CMQC.MQOO_OUTPUT);
+        } catch (MQException e) {
+            log.error("Error creating topic" + e);
+            return null;
```
Returning null and handle error as generic when you handle the null at higher layer, we will be loosing information why this got failed. It s better if we could throw this exception to higher layer and handle over there.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java:
```diff
@@ -0,0 +1,134 @@
+    public void connect(MessageContext messageContext) throws ConnectException {
+        try {
+
+            log.info("Message received at ibm-mq connector");
```
log.debug();

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java:
```diff
@@ -0,0 +1,134 @@
+
+            log.info("Message received at ibm-mq connector");
+            connectionBuilder = new MQConnectionBuilder(messageContext);
+            config = connectionBuilder.getConfig();
```
We need to use utility methods, as mentioned for the queue.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java:
```diff
@@ -0,0 +1,134 @@
+            }
+
+            if (topic == null || mqMessage == null) {
+                log.error("Error in publishing message");
```
(topic == null || mqMessage == null), returning null and logging as generic error is not a good practice. It s better if we could catch the exception here ( where you return null ) and propagate as ConnectException and end the flow from here.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java:
```diff
@@ -0,0 +1,134 @@
+            if (topic == null || mqMessage == null) {
+                log.error("Error in publishing message");
+            } else {
+                log.info("topic initialized");
```
May be we should use debug log here.

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java:
```diff
@@ -0,0 +1,134 @@
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
```
Returning null and handle error as generic, we will be loosing information why this got failed when you handle the null at higher layer. It s better if we could throw this exception to higher layer and handle over there.

djkevincr replied:
log.info("Error creating mq message" , e);

djkevincr commented on src/main/java/org/wso2/carbon/esb/connector/MQReportConstants.java:
```diff
@@ -0,0 +1,40 @@
+ * Constants for report message
+ */
+public class MQReportConstants {
+    public static final String MQFB_EXPIRATION = "Feedback indicating that the message was discarded because it had not been removed from the destination queue before its expiry time had elapsed";
```
Since we point a separate reply queue, handle this separately, we do not need to decode this message codes. Please remove this class.

djkevincr commented on src/test/java/org/wso2/carbon/connector/ibmmqIntegrationTest.java:
```diff
@@ -0,0 +1,50 @@
+/**
+ * Sample integration test
+ */
+public class ibmmqIntegrationTest extends ConnectorIntegrationTestBase {
```
IbmmqIntegrationTest

djkevincr commented on src/main/resources/publishqueue/component.xml:
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
com.ibm.mq.allclient.jar, com.ibm.mq.allclient.jar, com.ibm.mq.allclient.jar these jar files should be remove. Please remove all the proprietary IBM binary files from the PR.

djkevincr commented:
Please note you should remove all the proprietary IBM binary files committed from the PR. 
com.ibm.mq.allclient.jar, com.ibm.mq.allclient.jar, com.ibm.mq.allclient.jar

malakasilva commented on src/main/java/org/wso2/carbon/esb/connector/MQConfiguration.java:
```diff
@@ -0,0 +1,398 @@
+
+    MQConfiguration(MessageContext msg) {
+
+        if (msg.getProperty(MQConstants.PORT) != null) {
```
Can you set the default value first. Can reduce number of lines.

malakasilva replied:
EG:
 +        this.port = 1414;
 +        if (msg.getProperty(MQConstants.PORT) != null) {
 +            this.port = Integer.valueOf((String) msg.getProperty(MQConstants.PORT));
 +        }

malakasilva commented on src/main/java/org/wso2/carbon/esb/connector/MQConfiguration.java:
```diff
@@ -0,0 +1,398 @@
+    private int reconnectTimeout;
+    private List<String> channelList = new ArrayList<>();
+
+    MQConfiguration(MessageContext msg) {
```
Give a proper name for msg.

malakasilva commented on src/main/java/org/wso2/carbon/esb/connector/MQConfiguration.java:
```diff
@@ -0,0 +1,398 @@
+        }
+
+        String messageTypeString = (String) msg.getProperty(MQConstants.MESSAGE_TYPE);
+        if (messageTypeString != null) {
```
No point doing this. Can you directly assign.

hasithajayasundara committed:
Added changes for PR comments

hasithajayasundara committed:
Delete com.ibm.mq.allclient.jar

hasithajayasundara committed:
Delete fscontext.jar

hasithajayasundara committed:
Delete providerutil.jar

hasithajayasundara committed:
Merge pull request #3 from hasithajayasundara/ibmmqconnector

Ibmmqconnector

hasithajayasundara committed:
Resolved connection pool conflicts

hasithajayasundara committed:
Java doc comment updated

hasithajayasundara committed:
Error logging removed

hasithajayasundara committed:
Error logging removed from other classes

hasithajayasundara committed:
removed unwanted comments and debug added

riyafa commented:
What if we have only a few meaningful commits in the PR? You may squash commits in the following manner:
https://www.devroom.io/2011/07/05/git-squash-your-latests-commits-into-one/

hasithajayasundara committed:
Topic and queue manager seperate connections combined

hasithajayasundara committed:
Connection pool modified for use default connection pool

hasithajayasundara committed:
Readme.md modified

hasithajayasundara committed:
Removed unnecessary methods and readme updated.

hasithajayasundara committed:
readme updated

hasithajayasundara committed:
Removed groupId parameter and readme updated

hasithajayasundara committed:
Readme updated

hasithajayasundara committed:
Key to the connection pool changed

hasithajayasundara committed:
Updated readme.md

hasithajayasundara committed:
Performed a load test using jmeter and adjusted the code according to the result

Code cleaned up

changed method synchronization

Removed ConcurrentHashmap

Added Queue Manager disconnect method

Removed producerType configuration parameter

Removed keystore parameters

Init.xml changed for removed ssl parameters

Added report replyToQueue and repalyToQueueManager parameters for report messages

Message Type parameter changed
